### ISSUE TITLE
Analyser CS9236 causes IDE0004

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#78139]
**Brief description:**
When the `CS9236` analyser ("Compiling requires binding the lambda expression at least 100 times") is triggered, we can supply type hints to avoid the problem.

However that causes `IDE0004` "Cast is redundant".

So solving `CS9236` requires both a type hint AND a suppression of `IDE0004`.

**Languages applicable:**

c#

**Additional information:**

It would be nice if the one analyser automatically suppressed the other. Is that possible?

**Comment by huoyaoyuan** [comment]
It would be hard. CS9236 is a core compiler error (not analyzer), it means the compiler reaches its limitation during actual work.

In contrast, IDE0004 is an analyzer at outer layer. It only analyzes the semantic of code, not the effort required to get it compiled.

The core compiler is not aware of IDE1004 analyzer. The analyzer can't replicate the work of core compiler. You have to address them separately.

**Comment by CyrusNajmabadi** [comment]
> is triggered, we can supply type hints to avoid the problem.

The better way to handle this is to write an explicit return type for your lambda.  so `ret_type (X param1, Y param2 => ...`

You don't need a cast to address the issue here :)

**Comment by CyrusNajmabadi** [comment]
That not.  In your case you are casting the lambda. 

I'm saying that you should write a lambda with an actual return type and argument types. Like I showed in my post.

**Comment by huoyaoyuan** [comment]
No, you can explicitly specify the types within the lambda expression:

```csharp
foo.Sum(x =>
    x.Bar.Sum(int (Baz y) =>
        y.A + y.B));
```
